components/tabs/src/storage.rs

```diff
@@ -571,6 +578,44 @@ mod tests {
         );
     }
     #[test]
+    fn test_utf8_safe_title_trim() {
+        let mut storage = TabsStorage::new_with_mem_path("test_prepare_local_tabs_for_upload");
+        assert_eq!(storage.prepare_local_tabs_for_upload(), None);
+        storage.update_local_state(vec![RemoteTab {
+            title: "😍".repeat(MAX_TITLE_CHAR_LENGTH + 10), // Fill a string more than max
+            url_history: vec!["https://foo.bar".to_owned()],
+            icon: None,
+            last_used: 0,
+        }, RemoteTab {
+            title: "を".repeat(MAX_TITLE_CHAR_LENGTH + 5), // Fill a string more than max
+            url_history: vec!["https://foo_jp.bar".to_owned()],
+            icon: None,
+            last_used: 0,
+        }]);
+        let mut truncated_title = "😍".repeat(MAX_TITLE_CHAR_LENGTH - 1);
+        let mut truncated_jp_title = "を".repeat(MAX_TITLE_CHAR_LENGTH - 1);
+        truncated_title.push('\u{2026}');
+        truncated_jp_title.push('\u{2026}');
```

Comment:
I think you should check that the truncated titles are exactly `MAX_TITLE_CHAR_LENGTH` bytes here to makes sure you're testing the right thing.

Reply:
Given that we now truncate "correctly" we might not be able to exactly get the `MAX_TITLE_CHAR_LENGTH` since if something was `513` but 2 bytes then we'd get `511` if i'm not mistaken. Though I should add some extra tests to validate we are within the proper bytes